pkg/kubectl/cmd/rollout/rollout_status.go

```diff
@@ -133,18 +134,24 @@ func RunStatus(f *cmdutil.Factory, cmd *cobra.Command, out io.Writer, args []str
 	}
 
 	// if the rollout isn't done yet, keep watching deployment status
-	kubectl.WatchLoop(w, func(e watch.Event) error {
-		// print deployment's status
-		status, done, err := statusViewer.Status(cmdNamespace, info.Name)
-		if err != nil {
-			return err
-		}
-		fmt.Fprintf(out, "%s", status)
-		// Quit waiting if the rollout is done
-		if done {
-			w.Stop()
-		}
-		return nil
+	intr := interrupt.New(nil, w.Stop)
+	defer intr.Close()
+	intr.Run(func() error {
+		_, err := watch.Until(0, w, func(e watch.Event) (bool, error) {
+			// print deployment's status
+			status, done, err := statusViewer.Status(cmdNamespace, info.Name)
+			if err != nil {
+				return false, err
+			}
+			fmt.Fprintf(out, "%s", status)
+			// Quit waiting if the rollout is done
+			if done {
+				w.Stop()
```

Comment:
`w.Stop` will be called when `intr.Run()` exits, as part of its deferred called to `h.Close()`, so you should be able to remove this line, right?